File: Maynghien.Infrastructure/Models/Request/SearchRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MayNghien.Infrastructure.Request.Base
{
    public class SearchRequest
    {
        public List<Filter>? Filters { get; set; }

        public SortByInfo? SortBy { get; set; }
        [Required]
        [Range(0, int.MaxValue)]
        public int? PageIndex { get; set; }

        [Required]
        [Range(1, int.MaxValue)]
        public int? PageSize { get; set; }
    }
}

File: OKR.UI/src/components/maynghien/BaseModels/SearchRequest.ts
// @ts-ignore
import {Filter} from '../BaseModels/Filter';
// @ts-ignore
import {SortByInfo} from "../BaseModels/SortByInfo";


export class SearchRequest {
    
    PageIndex: number | undefined;
    PageSize:number |undefined;
    filters:Filter[]|undefined;
    SortBy:SortByInfo|undefined;
}